mods/cnc/maps/twist-of-fate/twist-of-fate.lua

```diff
@@ -0,0 +1,328 @@
+end
+
+CheckObjectives = function()
+	if Nod.HasNoRequiredUnits() then GDI.MarkCompletedObjective(EliminateAllNod) end
+	if GDI.HasNoRequiredUnits() then GDI.MarkFailedObjective(EliminateAllNod) end
+	Trigger.AfterDelay(50, CheckObjectives)
+end
+
+CompleteCaptureCommCenterObjective = function()
+	GDI.MarkCompletedObjective(CaptureCommCenter)
+	if not NodCYard.IsDead and NodCYard.Owner == Nod then
+		Trigger.Clear(NodCYard, "OnDamaged")
+		Trigger.AfterDelay(1, function()
+			RepairBuilding(Nod, NodCYard, 0.75)
+			NodCYard.StartBuildingRepairs()
+		end)
+	end
+	Media.DisplayMessage(UserInterface.Translate("communications-center-captured-sams-located"))
+	local activeSams = Nod.GetActorsByType("sam")
+	local miniCams = { }
+	if #activeSams > 0 then
+		Utils.Do(activeSams, function(s)
+			table.insert(miniCams, Actor.Create("camera.mini", true, { Owner = GDI, Location = s.Location }))
+			table.insert(miniCams, Actor.Create("camera.mini", true, { Owner = GDI, Location = CPos.New(s.Location.X + 1, s.Location.Y) }))
+		end)
+		Trigger.AfterDelay(1, function()
+			Utils.Do(miniCams, function(c)
+				c.Destroy()
+			end)
+		end)
+	end
+end
+
+WorldLoaded = function()
+
+	Camera.Position = DefaultCameraPosition.CenterPosition
+	GDI = Player.GetPlayer("GDI")
+	Nod = Player.GetPlayer("Nod")
+
+	if Difficulty == "hard" then
+		ProduceBuildingsDelay = 50
+		ProduceUnitsDelay = 10
+		ProductionCooldownSeconds = 30
+		MinHarvs = 3
+		AstkDelay = 180
+		NukeDelay = 900
+		ProcUpgrade = "AIHProcUpgrade"
+		Nod.Resources = 3000
+		GDI.Cash = 3000
+		MCVReinf = { "mcv" }
+	end
+
+	if Difficulty == "normal" then
+		ProduceBuildingsDelay = 70
+		ProduceUnitsDelay = 15
+		ProductionCooldownSeconds = 55
+		MinHarvs = 3
+		AstkDelay = 220
+		NukeDelay = 1200
+		Nod.Resources = 3000
+		GDI.Cash = 4000
+		ProcUpgrade = "AINProcUpgrade"
+		MCVReinf = { "mtnk", "mcv" }
+	end
+
+	if Difficulty == "easy" then
+		ProduceBuildingsDelay = 100
+		ProduceUnitsDelay = 35
+		ProductionCooldownSeconds = 85
+		MinHarvs = 2
+		AstkDelay = 250
+		NukeDelay = 1500
+		Nod.Resources = 2000
+		GDI.Cash = 5000
+		MCVReinf = { "mtnk", "mtnk", "mcv", "mtnk" }
+		RT1 = { "jeep", "jeep", "apc" }
+		RT2 = { "mtnk", "msam" }
+		RT3 = { "htnk" }
+		ReinforceTeams = { RT1, RT2, RT3 }
+		ReinforceDelay = 240
+		Actor137.Teleport(CPos.New(57,6))
+		Actor203.Destroy()
+	end
+
+	InitObjectives(GDI)
+	ClearPath = AddPrimaryObjective(GDI, "clear-path")
+	Trigger.AfterDelay(DateTime.Seconds(5), function()
+		EliminateAllNod = AddPrimaryObjective(GDI, "eliminate-nod")
+	end)
+
+	InitAI()
+	CheckObjectives()
+
+	Flare = Actor.Create("flare", true, { Owner = GDI, Location = DefaultFlareLocation.Location })
+
+	Trigger.AfterDelay(1,function()
+		AmbushTeam = Map.ActorsInBox(Map.CenterOfCell(CPos.New(46,5)), Map.CenterOfCell(CPos.New(60,53)), function(a) return a.Owner == Nod and a.Type ~= "stnk" end)
+		local neutralHarv = Player.GetPlayer("Neutral").GetActorsByType("harv")
+		neutralHarv[1].Destroy()
```

Comment:
I assume that destroys the harvester spawned by the refinery. This should instead be done by specifying `FreeActor: false` on the actor definition in map.yaml, see https://github.com/OpenRA/OpenRA/blob/621c85059ec92f550ad25044063f73ed326ae8fa/mods/cnc/maps/nod07a/map.yaml#L538-L541 for an example.